web/src/components/questions/RegistrationCertificateQuestion.tsx

```diff
@@ -0,0 +1,104 @@
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact SUSE LLC.
+ *
+ * To contact SUSE LLC about this file by physical or electronic mail, you may
+ * find current contact information at www.suse.com.
+ */
+
+import React from "react";
+import spacingStyles from "@patternfly/react-styles/css/utilities/Spacing/spacing";
+import {
+  Content,
+  DescriptionList,
+  DescriptionListDescription,
+  DescriptionListGroup,
+  DescriptionListTerm,
+  Stack,
+  StackItem,
+} from "@patternfly/react-core";
+import { Popup } from "~/components/core";
+import { AnswerCallback, Question } from "~/types/questions";
+import QuestionActions from "~/components/questions/QuestionActions";
+import { _ } from "~/i18n";
+
+type QuestionDataProps = {
+  label: string;
+  value?: string;
+};
+
+function QuestionData({ label, value }: QuestionDataProps): React.ReactNode | null {
+  if (!value) return null;
+
+  return (
+    <DescriptionListGroup>
+      <DescriptionListTerm>{label}</DescriptionListTerm>
+      <DescriptionListDescription>{value}</DescriptionListDescription>
+    </DescriptionListGroup>
+  );
+}
+
+/**
+ * Component to ask for trusting a self signed certificate.
+ *
+ * @param question - the question to be answered
+ * @param answerCallback - the callback to be triggered on answer
+ */
+export default function RegistrationCertificateQuestion({
+  question,
+  answerCallback,
+}: {
+  question: Question;
+  answerCallback: AnswerCallback;
+}): React.ReactNode {
+  const actionCallback = (option: string) => {
+    question.answer = option;
+    answerCallback(question);
+  };
+
+  return (
+    <Popup isOpen title={_("Registration certificate")}>
+      <Stack hasGutter>
+        <StackItem>
+          <Content component="p" isEditorial>
+            {question.text}
+          </Content>
+        </StackItem>
+        <StackItem>
+          <DescriptionList isHorizontal isCompact className={spacingStyles.mxLg}>
+            <QuestionData label={_("URL")} value={question.data.url} />
+            <QuestionData label={_("Issuer")} value={question.data.issuer_name} />
+            <QuestionData label={_("Issue date")} value={question.data.issue_date} />
+            <QuestionData label={_("Expiration date")} value={question.data.expiration_date} />
+            <QuestionData label={_("SHA1 fingerprint")} value={question.data.sha1_fingerprint} />
+            <QuestionData
+              label={_("SHA256 fingerprint")}
+              value={question.data.sha256_fingerprint}
+            />
+          </DescriptionList>
+        </StackItem>
+      </Stack>
+      <Popup.Actions>
+        <QuestionActions
+          actions={question.options}
+          defaultAction={question.defaultOption}
```

Comment:
NP: Probably not in the scope of this PR, but reveleaded here.

I believe we have to look a way for not having a "primary" action but just the focused one for prompts like this where Agama shouldn't use a call to action. Agama should just prevent users trusting in the self certificate by accident focusing the Reject by default, but not encouraging/inviting users to reject it by using a primary action look&feel.